src/main/java/at/ac/tuwien/kr/alpha/solver/heuristics/BerkMin.java

```diff
@@ -189,11 +193,24 @@ private void pushToStack(NoGood noGood) {
 	}
 
 	private void incrementActivityCounter(int literal) {
-		activityCounters.compute(atomOf(literal), (k, v) -> (v == null ? DEFAULT_ACTIVITY : v) + 1);
+		int atom = atomOf(literal);
+		if (isAtomChoicePoint.test(atom)) {
+			activityCounters.compute(atom, (k, v) -> (v == null ? DEFAULT_ACTIVITY : v) + 1);
+		}
+		// TODO: check performance
+		// note that here (and in incrementSignCounter) we only count atoms that are
+		// choice points, which might affect performance.
+		// alternative approaches:
+		// 1. count everything and from time to time do a garbage collection (i.e.
+		// remove atoms that are no choice points)
+		// 2. make check cheaper, e.g. by using dedicated atom IDs (e.g. even
+		// integers for rule bodies, uneven for other atoms)
 	}
 	
 	private void incrementSignCounter(Integer literal) {
-		signCounters.compute(atomOf(literal), (k, v) -> (v == null ? DEFAULT_SIGN_COUNTER : v) + 1);
+		if (isAtomChoicePoint.test(atomOf(literal))) {
+			signCounters.compute(literal, (k, v) -> (v == null ? DEFAULT_SIGN_COUNTER : v) + 1);
```

Comment:
Oh, I understand. :sweat_smile:  I even failed to see the very commit highlighting the change...
Sometimes "better safe than sorry" fires back :blush: